source/v4/saml.yaml

```diff
@@ -0,0 +1,182 @@
+          schema:
+            type: string
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+  /saml/certificate/idp:
+    post:
+      tags:
+        - SAML
+      summary: Upload IDP certificate
+      description: |
+        Upload the IDP certificate to be used with your SAML configuration. This will also set the filename for the IdpCertificateFile setting in your `config.json`.
+        ##### Permissions
+        Must have `manage_system` permission.
+      parameters:
+        - name: certificate
+          in: formData
+          description: The IDP certificate file
+          required: true
+          type: file
+      responses:
+        '200':
+          description: SAML certificate upload successful
+          schema:
+            $ref: '#/definitions/StatusOK'
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+    delete:
+      tags:
+        - SAML
+      summary: Remove IDP certificate
+      description: |
+        Delete the current IDP certificate being used with your SAML configuration. This will also disable SAML on your system as this certificate is required for SAML.
+        ##### Permissions
+        Must have `manage_system` permission.
+      responses:
+        '200':
+          description: SAML certificate delete successful
+          schema:
+            $ref: '#/definitions/StatusOK'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+  /saml/certificate/public:
+    post:
+      tags:
+        - SAML
+      summary: Upload public certificate
+      description: |
+        Upload the public certificate to be used with your SAML configuration. This will also set the filename for the PublicCertificateFile setting in your `config.json`.
+        ##### Permissions
+        Must have `manage_system` permission.
+      parameters:
+        - name: certificate
+          in: formData
+          description: The public certificate file
+          required: true
+          type: file
+      responses:
+        '200':
+          description: SAML certificate upload successful
+          schema:
+            $ref: '#/definitions/StatusOK'
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+    delete:
+      tags:
+        - SAML
+      summary: Remove public certificate
+      description: |
+        Delete the current public certificate being used with your SAML configuration. This will also disable SAML on your system as this certificate is required for SAML.
```

Comment:
Definitely not true as this certificate is to be use to verify the if the SAML response is a valid response, you can have your SAML configured without the need the verification step